bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractBatchProcessingWorker.php

```diff
@@ -235,22 +267,63 @@ public function prepareDataForIndex(IndexableInterface $object)
                 ]);
 
                 $jsonLastError = \json_last_error();
+                $generalErrors = [];
                 if ($jsonLastError !== JSON_ERROR_NONE) {
-                    throw new \Exception("Could not encode product data for updating index. Json encode error code was {$jsonLastError}, ObjectId was {$subObjectId}.");
+                    $e = new \Exception("Could not encode product data for updating index. Json encode error code was {$jsonLastError}, ObjectId was {$subObjectId}.");
+                    $event = new PreprocessErrorEvent($e);
+                    $this->eventDispatcher->dispatch(IndexServiceEvents::GENERAL_PREPROCESSING_ERROR, $event);
+                    if ($event->isPropagationStopped()) {
```

Comment:
we should not use stopPropagation for this but add an additional flag on the event to decide if exception should be thrown or not